src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java

```diff
@@ -127,6 +147,19 @@ public void stop() {
     // set task logging to default
     SnowflakeSinkTask.setTotalTaskCreationCount(-1);
     setupComplete = false;
+
+    // retry closing streaming client
+    int retryCount = 0;
+    while (this.isStreamingIngestion
+        && retryCount < this.closeConnectorRetryCount
+        && !IngestSdkProvider.clientManager.closeStreamingClient()) {
+      retryCount++;
+      LOGGER.debug(
```

Comment:
note: closing the client here is best effort - we log the closure failures in clientManager, and ultimately orphaned clients will not charge the customer and should be gc'ed